### ISSUE TITLE
Why re-invent the wheel?

### ISSUE THREAD

**Issue by v3ss0n** [vkryachko/Vase#4]
There is tornado , and it support asyncio too , may be you may want to base on it.
http://tornado.readthedocs.org/en/latest/asyncio.html

**Comment by harvimt** [comment]
Can you do something like

```
@app.route('/route/to/here')
@asyncio.coroutine
def foo(request):
    yield from ...
    return 'whatever...'
```

the `yield from ...` being the main thing.

Tornado may mechanically interface with the event loop, but unless it
integrates with the interface up front, it's not nearly as useful.

It looks like Tornado has it's own concept of a `Future` that can be
converted between asyncio's version, but it isn't "native"

(In fairness I haven't used Tornado at all, but it always looked super
convoluted to me.)

Though Vase is more of a proof-of-concept that anything else, I admire it's
simplicity + native `asyncio` support.

**Comment by vkryachko** [comment]
Hey @v3ss0n, thanks for the input. I don't think tornado had asyncio support when I started the project. And even if it did, I would still not use it as the main goal of the project was to learn and play around with asyncio, and I was really interested in implementing http and websocket protocol myself. As to "why" I did it, because I wanted to and had some spare time to devote to it :)

@harvimt I am glad you like it and find it useful
